src/core/jit/model.rs

```diff
@@ -304,11 +302,21 @@ impl Debug for Flat {
 #[derive(Clone, Debug)]
 pub struct Nested<Input>(Vec<Field<Nested<Input>, Input>>);
 
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 pub struct OperationPlan<Input> {
     flat: Vec<Field<Flat, Input>>,
     operation_type: OperationType,
     nested: Vec<Field<Nested<Input>, Input>>,
+    errors: Arc<Mutex<Vec<LocationError<Error>>>>,
+    pub index: Arc<Index>,
```

Comment:
Drop `errors` and `index`. Lets attempt to make `OperationPlan` remains immutable.